src/Assignments/ProblemIframe.tsx

```diff
@@ -139,39 +128,35 @@ export const ProblemIframe: React.FC<ProblemIframeProps> = ({
             return;
         }
 
-        body.onresize = recalculateHeight;
-
         // HTMLCollectionOf is not iterable by default in Typescript.
         // const forms = iframeDoc.getElementsByTagName('form');
         // _.forEach(forms, form => form.addEventListener('submit', hijackFormSubmit));
         insertListener();
-
-        console.log('Checking MathJax...');
-        const MathJax = (iframeRef.current?.contentWindow as any)?.MathJax;
-        if (MathJax !== undefined) {
-            console.log('Found MathJax!');
-            MathJax.Hub?.Register?.StartupHook('End', function () {
-                console.log('Recalculating because MathJax has finished computing.');
-                recalculateHeight();
-            });
-        } else {
-            console.log('Couldn\'t find MathJax!');
-        }
-
         setLoading(false);
     };
 
     return (
         <>
             { loading && <Spinner animation='border' role='status'><span className='sr-only'>Loading...</span></Spinner>}
             {error && <div>{error}</div>}
-            <iframe
+            <IframeResizer
+                // Using onInit instead of ref because:
+                // ref never get's set and a warning saying to use `forwardRef` comes up in the console
+                // Using forwardRef does not give you access to the iframe, rather it gives you access to 3 or 4 methods and properties (like `sendMessage`)
+                onInit={(iframe: IFrameComponent) => {
+                    // TODO do we need to unset the iframeref? As of right now it should not be required since it is always present within the component
+                    // If using dom elements the useRef is "Read Only", however I want control!
+                    (iframeRef as any).current = iframe;
```

Comment:
I'd say, since we can hook-in via the event handlers, we don't need a ref for anything.

Reply:
upcoming changes for save and submit use the ref for grabbing the form element and GeoGebra-related scripts in the iframe

Reply:
have not tested the modified save/submit with these changes though...